torch/csrc/api/src/nn/modules/adaptive.cpp

```diff
@@ -106,14 +132,18 @@ ASMoutput AdaptiveLogSoftmaxWithLossImpl::forward(const Tensor& input, const Ten
   TORCH_CHECK(
     used_rows == batch_size,
     "Target values should be in [0, ", options.n_classes() - 1, "], "
-    "but values in range [", target.min().item().toDouble(), ", ", target.max().item().toDouble(), "] "
+    "but values in range [", new_target.min().item().toDouble(), ", ", new_target.max().item().toDouble(), "] "
     "were found. ");
 
-  const Tensor head_output = head(input);
+  const Tensor head_output = head(new_input);
   const Tensor head_logprob = F::log_softmax(head_output, 1);
   output += head_logprob.gather(1, gather_inds.unsqueeze(1)).squeeze();
   const double loss = (-output).mean().item().toDouble();
 
+  if (targ_dim == 0) {
+    output = output.squeeze(0);
+  }
+
```

Comment:
Do we want the output to be a scalar here? This produces an output that looks something like:

`tensor(-4.5369, grad_fn=<SqueezeBackward1>)`

whereas the same input that isn't 1D/0D yields 

`tensor([-4.5369], grad_fn=<AddBackward0>)`

Is this okay?

Reply:
Yeah, we want a tensor scalar output when a tensor scalar is passed in for the target.